compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1996,30 +1996,14 @@ impl<'a, 'tcx> InferCtxtPrivExt<'a, 'tcx> for InferCtxt<'a, 'tcx> {
                 // begin with in those cases.
                 if self.tcx.lang_items().sized_trait() == Some(trait_ref.def_id()) {
                     if !self.is_tainted_by_errors() {
-                        self.emit_inference_failure_err(
-                            body_id,
-                            span,
-                            subst,
-                            vec![],
-                            ErrorCode::E0282,
-                        )
-                        .emit();
+                        self.emit_inference_failure_err(body_id, span, subst, ErrorCode::E0282)
+                            .emit();
                     }
                     return;
                 }
 
-                let impl_candidates = self
-                    .find_similar_impl_candidates(trait_ref)
-                    .into_iter()
-                    .map(|candidate| candidate.trait_ref)
-                    .collect();
```

Comment:
Could we make a note to see if we can check whether this snippet is useful to modify `err` in some way? I fear we must have had a suggestion regression at some point.